src/users/index.ts

```diff
@@ -18,35 +46,151 @@ export interface Users {
    * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
   getAssignedRoles: (userId: string) => Promise<Record<string, Role>>;
+
+  db: DBUsers;
+  oidc: OIDCUsers;
+}
+
+/** Operations supported for namespaced 'db' users.*/
+export interface DBUsers extends UsersBase {
   /**
-   * Assign roles to a user.
+   * Retrieve the roles assigned to a user.
    *
-   * @param {string | string[]} roleNames The name or names of the roles to assign.
-   * @param {string} userId The ID of the user to assign the roles to.
-   * @returns {Promise<void>} A promise that resolves when the roles are assigned.
+   * @param {string} userId The ID of the user to retrieve the assigned roles for.
+   * @returns {Promise<Record<string, Role>>} A map of role names to their respective roles.
    */
-  assignRoles: (roleNames: string | string[], userId: string) => Promise<void>;
+  getAssignedRoles: (userId: string, opts?: GetAssingedRolesOptions) => Promise<Record<string, Role>>;
+
+  create: (userId: string) => Promise<string>;
+  delete: (userId: string) => Promise<boolean>;
+  rotateKey: (userId: string) => Promise<string>;
+  activate: (userId: string) => Promise<boolean>;
+  deactivate: (userId: string) => Promise<boolean>;
+  byName: (userId: string) => Promise<UserDB>;
+  listAll: () => Promise<UserDB[]>;
+}
+
+type GetAssingedRolesOptions = {
+  includePermissions?: boolean;
+};
```

Comment:
Can we do the following here?
- Move this to `src/roles/types.ts`
- Export the type from that file
- Fix the `Assinged` typo to `Assigned`

Cheers! 😁

Reply:
Totes 💯